apiserver/facades/agent/uniter/goal-state_test.go

```diff
@@ -0,0 +1,336 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package uniter_test
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facades/agent/uniter"
+	"github.com/juju/juju/apiserver/params"
+	apiservertesting "github.com/juju/juju/apiserver/testing"
+	"github.com/juju/juju/juju/testing"
+	"github.com/juju/juju/state"
+	jujufactory "github.com/juju/juju/testing/factory"
+)
+
+// uniterSuite implements common testing suite for all API
+// versions. It's not intended to be used directly or registered as a
+// suite, but embedded.
+type uniterGoalStateSuite struct {
+	testing.JujuConnSuite
+
+	authorizer apiservertesting.FakeAuthorizer
+	resources  *common.Resources
+	uniter     *uniter.UniterAPI
+
+	machine0      *state.Machine
+	machine1      *state.Machine
+	machine2      *state.Machine
+	wordpress     *state.Application
+	wpCharm       *state.Charm
+	mysql         *state.Application
+	wordpressUnit *state.Unit
+	mysqlUnit     *state.Unit
+
+	meteredService *state.Application
```

Comment:
Why are all these extra suite attributes here? There's not used nor required in any test on this suite. We need to only define and use the minimum required data model to support the tests.